### PR TITLE
Feat: Pet Skin and Glamour Anvil and Color changes

### PR DESCRIPTION
**Pull request by Zintixx**
3 main updates to this: Item Extraction, Pet Skin Scrolls, and Color changes
## Item Extraction
Implemented Item Extraction aka Glamour Anvils
Parsing `itemextraction.xml` for needed data.
Adjusted `Item` to grab the correct amount of glamour forges remaining

## Pet Skin Scrolls
Implemented Pet Skin scrolls (ability to make a pet skin badge).

## Color Changes
Now correctly writing color for items.
`ItemMetadataStorage.GetEquipColor` shows how a color is determined for an item.
Revised `EquipColor` to contain `MixedColor`, Color Index, and Color Palette.
Added `MixedColor`. This struct _ONLY_ has color to it.
Removed AppearanceFlag in `Item`. This turned out to be Color Palette.

### Other
Added all the Gacha shops
Implemented HatData. This contains the position information for hats.
Revised `ShopHandler` to send each shop item as it's own packet. This is normally how it is behaved in game. With the changes needed on Gacha shops, it required sending each packet individually.
Now adding in items received from gacha capsules the dismantle information. Also reflected in `WriteItem`
Transparency Badge functionality now works

### CODE REVIEW

File: GameDataParser/Parsers/ColorPaletteParser.cs [hunk 1]
```diff
@@ -48,11 +48,15 @@ protected override List<ColorPaletteMetadata> Parse()
                             byte[] tertiaryBytes = BitConverter.GetBytes(tertiary);
                             Array.Reverse(tertiaryBytes);
 
-                            EquipColor newColor = EquipColor.Custom(
+                            int paletteColor = Convert.ToInt32(colorNode.Attributes["palette"].Value, 16);
+                            byte[] paletteBytes = BitConverter.GetBytes(paletteColor);
+                            Array.Reverse(paletteBytes);
+
+                            MixedColor newColor = MixedColor.Custom(
                     Color.Argb(primaryBytes[0], primaryBytes[1], primaryBytes[2], primaryBytes[3]),
```
Comment (AlanMorel) [inline GameDataParser/Parsers/ColorPaletteParser.cs:56]: Optional but could be cool to have a method called FromBytes in Color that just takes in bytes and does this whole index thing there

**Review by AlanMorel — CHANGES_REQUESTED** [review]

**Review by AlanMorel — APPROVED** [review]

File: MapleServer2/Constants/RecvOp.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public enum RecvOp : ushort
         STATE_CONSUME_EP = 0x0033,
         STATE_FALL_DAMAGE = 0x0034,
         GUIDE_OBJECT_SYNC = 0x0035,
-        REQUEST_SET_CRAFT_MODE = 0x0036,
+        REQUEST_SET_BUILD_MODE = 0x0036,
```
Comment (angelotadeucci) [inline MapleServer2/Constants/RecvOp.cs:53]: Rename `SET_CRAFT_MODE` on **SendOp** too

File: MapleServer2/PacketHandlers/Game/RequestGemEquipmentHandler.cs [hunk 3]
```diff
@@ -88,5 +92,18 @@ private static void HandleUnequipItem(GameSession session, PacketReader packet)
...
+            byte slot = packet.ReadByte();
+            byte[] transparencyBools = packet.Read(10);
+
+            Item item = session.Player.Badges[slot];
+
+            item.TransparencyBadgeBools = transparencyBools;
+
+            session.FieldManager.BroadcastPacket(GemPacket.EquipItem(session, item));
+        }
+
```
Comment (angelotadeucci) [inline MapleServer2/PacketHandlers/Game/RequestGemEquipmentHandler.cs:107]: blank line

File: MapleServer2/Packets/ItemExtractionPacket.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
+﻿using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Types;
+
+namespace MapleServer2.Packets
+{
+
```
Comment (angelotadeucci) [inline MapleServer2/Packets/ItemExtractionPacket.cs:7]: blank line

File: MapleServer2/Packets/PetSkinPacket.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Types;
+
+namespace MapleServer2.Packets
+{
+
```
Comment (angelotadeucci) [inline MapleServer2/Packets/PetSkinPacket.cs:7]: blank line

**Review by angelotadeucci — APPROVED** [review]

**Pull request merged**